### PR TITLE
JITMS: Add the ability to parse lists

### LINKED ISSUE #8948: Enhancement: Include "plan expiring" banner within plugin and calypso
When you have a Jetpack plan which is expiring, there's no indication within the plugin, nor within Calypso (except for on the "Manage Purchases" view).

Aside from the email that's sent out, we should include a banner in:
1) Calypso 
2) Jetpack dashboard within wp-admin
3) Main dashboard of wp-admin

I want to implement the multiline JITM that myself and @withinboredom built out **in order to provide a more personalized experience** to reduce churn. Based on whatever plan the user has, we should **list their top 3 most used features** that will be lost if the plan expires.

I realize that may be a bit of logic to write, so if that's too much, perhaps we add three generally popular features that they stand to lose (see mockup below).

**Example visual on the jetpack dashboard:**

plan-expiring-jitm

**The professional plan expiry JITM should read:** 
Line 1: "Your Jetpack Professional subscription expires in 1 month!"
Line 2: "Don't lose out on these great Professional features"

Bulleted list:
* most used feature
* 2nd most used feature
* 3rd most used feature

[link]Renew now[/link]

This way the JITM is much more personalized and will hopefully motivate folks to renew. And the multiline JITMs are already built, just not in use. 

@richardmuscat @nicoleckohler I'd like your thoughts on the personalization and verbiage. cc @bubel so we can stay consistent with the emails. cc @beaulebens who initially made the suggestion

### PR DESCRIPTION
**Pull request by withinboredom**
Fixes #8948

image

#### Changes proposed in this Pull Request:

* Adds the ability to display and parse a list array from wpcom

#### Testing instructions:

* Get an annual plan that expires in less than 30 days (may need SA assistance to change expiration date)
* Go to the dashboard
* See this jitm

#### Proposed changelog entry for your changes:
* If your plan will expire soon, we'll give you a heads up on your dashboard

### CODE REVIEW

**Comment by jeffgolenski** [comment]
I've modded the styles a bit so it's looking good on large and small screens now.

screen shot 2018-03-07 at 12 16 22 pm
screen shot 2018-03-07 at 12 16 47 pm


@withinboredom there's still some things I need from you since the technicality of the code is a little above my skillset.

**1) Change the title based on which plan the customer has.**

it currently reads: "Your Jetpack subscription expires in X days!"

We want it to read: "Your Jetpack [Plan name] Plan expires in X days!"

(We don't want any confusion between our plans and our jetpack subscription feature)

**2) Alter checklist items based on plan**

Personal plan should read: 

* Site backups & restores
* Spam filtering
* Priority support
* And more! (I added this to jetpack-jitm.js, but please feel free to clean up that code)

The "and more" is a link to the pricing page on jetpack.com that list all the features in each respective plan.

Visual example of what it should end up looking like:

screen shot 2018-03-07 at 1 21 16 pm


Premium plan should read:

* Site backups & restores
* Spam filtering & malware scanning
* Priority support
* And more!

Professional plan should read:

* 200+ Premium themes
* Real-time backups & restores
* On-demand malware scanning
* And more! 

**3) Make sure our tracking is in place:**

* add tracks to "and more" link in each list

We're going to need to see how many folks see this JITM and how many folks actually click renew and who dismisses it. However, since folks may renew via email messages or whatnot, I wonder if its possible to cross-ref that data to get a better picture of how successful our new JITM and renew emails are in comparison to overall paid churn.

**Comment by withinboredom** [comment]
Requires D10824-code and D10827-code to be merged.

**Comment by withinboredom** [comment]
**1) Title is dynamically changed based on number of days to/from expiration**
**2) List items are handled by the new `/plans` endpoint**
**3) Tracking is configured**

@jeffgolenski Please confirm everything looks correct on your end, and lets 🚢 it!

**Comment by jeffgolenski** [comment]
Per our conversation via slack: The lists being pulled in are a bit long and not as personalized as I'd like to see. Rob making adjustments.

File: _inc/jetpack-jitm.js [hunk 1]
```diff
@@ -9,7 +9,26 @@ jQuery( document ).ready( function( $ ) {
 			html += '<div class="jitm-banner__info">';
 			html += '<div class="jitm-banner__title">' + envelope.content.message + '</div>';
 			if ( envelope.content.description && envelope.content.description !== '' ) {
-				html += '<div class="jitm-banner__description">' + envelope.content.description + '</div>';
+				html += '<div class="jitm-banner__description">' + envelope.content.description;
+				if( envelope.content.list.length > 0 ) {
```
Comment (dereksmart) [inline _inc/jetpack-jitm.js]: missing whitespace after `if`

File: _inc/jetpack-jitm.js [hunk 2]
```diff
@@ -9,7 +9,26 @@ jQuery( document ).ready( function( $ ) {
...
+					html += '<ul class="banner__list">';
+					for( var i = 0; i < envelope.content.list.length; i++ ) {
```
Comment (dereksmart) [inline _inc/jetpack-jitm.js]: whitespace

File: _inc/jetpack-jitm.js [hunk 3]
```diff
@@ -9,7 +9,26 @@ jQuery( document ).ready( function( $ ) {
...
+
+						var text = envelope.content.list[ i ].item;
+
+						if( envelope.content.list[ i ].url ) {
```
Comment (dereksmart) [inline _inc/jetpack-jitm.js]: whitespace

**Comment by dereksmart** [comment]
Whitespace addressed, :shipit:

**Pull request merged**